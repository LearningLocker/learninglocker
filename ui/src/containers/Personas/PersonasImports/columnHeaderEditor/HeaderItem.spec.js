import React from 'react';
import { shallow } from 'enzyme';
import 'jest-enzyme';
import { fromJS } from 'immutable';
import {
  COLUMN_MBOX
} from 'lib/constants/personasImport';
import {
  HeaderItemComponent
} from './HeaderItem';

describe('HeaderItem', () => {
  it('should render', () => {
    const rendered = shallow(<HeaderItemComponent
      columnName="test1"
      columnStructure={fromJS({})}
      model={fromJS({
        _id: '1'
      })} />);

    expect(rendered).toMatchSnapshot();
  });

  it('should render with orderable and related column', () => {
    const rendered = shallow(<HeaderItemComponent
      columnName="test1"
      columnStructure={fromJS({
        1: {
          columnName: 1,
          columnType: COLUMN_MBOX
        }
      })}
      model={fromJS({
        _id: '1'
      })} />);

    expect(rendered).toMatchSnapshot();
  });

  it('should render with orderable when disabled', () => {
    const rendered = shallow(<HeaderItemComponent
      columnName="test1"
      columnStructure={fromJS({
        1: {
          columnName: 1,
          columnType: COLUMN_MBOX
        }
      })}
      model={fromJS({
        _id: '1'
      })}
      disabled="true" />);

    expect(rendered).toMatchSnapshot();
  });
});
