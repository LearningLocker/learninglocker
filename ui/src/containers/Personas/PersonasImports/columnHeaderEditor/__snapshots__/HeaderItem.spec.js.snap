// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderItem should render 1`] = `
<div>
  <h3>
    test1
  </h3>
  <div
    className="form-group"
  >
    <label
      htmlFor="1-test1-columnType"
    >
      Field type
    </label>
    <select
      className="form-control"
      id="1-test1-columnType"
      value=""
    >
      <option
        value=""
      >
        Nothing
      </option>
      <option
        key="COLUMN_NAME"
        value="COLUMN_NAME"
      >
        Persona name
      </option>
      <option
        key="COLUMN_MBOX"
        value="COLUMN_MBOX"
      >
        Identifier mbox
      </option>
      <option
        key="COLUMN_MBOXSHA1SUM"
        value="COLUMN_MBOXSHA1SUM"
      >
        Identifier mbox sha1sum
      </option>
      <option
        key="COLUMN_OPENID"
        value="COLUMN_OPENID"
      >
        Identifier open id
      </option>
      <option
        key="COLUMN_ACCOUNT_KEY"
        value="COLUMN_ACCOUNT_KEY"
      >
        Account home page
      </option>
      <option
        key="COLUMN_ACCOUNT_VALUE"
        value="COLUMN_ACCOUNT_VALUE"
      >
        Account name
      </option>
      <option
        key="COLUMN_ATTRIBUTE_DATA"
        value="COLUMN_ATTRIBUTE_DATA"
      >
        Attribute value
      </option>
    </select>
  </div>
</div>
`;

exports[`HeaderItem should render with orderable and related column 1`] = `
<div>
  <h3>
    test1
  </h3>
  <div
    className="form-group"
  >
    <label
      htmlFor="1-test1-columnType"
    >
      Field type
    </label>
    <select
      className="form-control"
      id="1-test1-columnType"
      value=""
    >
      <option
        value=""
      >
        Nothing
      </option>
      <option
        key="COLUMN_NAME"
        value="COLUMN_NAME"
      >
        Persona name
      </option>
      <option
        key="COLUMN_MBOX"
        value="COLUMN_MBOX"
      >
        Identifier mbox
      </option>
      <option
        key="COLUMN_MBOXSHA1SUM"
        value="COLUMN_MBOXSHA1SUM"
      >
        Identifier mbox sha1sum
      </option>
      <option
        key="COLUMN_OPENID"
        value="COLUMN_OPENID"
      >
        Identifier open id
      </option>
      <option
        key="COLUMN_ACCOUNT_KEY"
        value="COLUMN_ACCOUNT_KEY"
      >
        Account home page
      </option>
      <option
        key="COLUMN_ACCOUNT_VALUE"
        value="COLUMN_ACCOUNT_VALUE"
      >
        Account name
      </option>
      <option
        key="COLUMN_ATTRIBUTE_DATA"
        value="COLUMN_ATTRIBUTE_DATA"
      >
        Attribute value
      </option>
    </select>
  </div>
</div>
`;

exports[`HeaderItem should render with orderable when disabled 1`] = `
<div>
  <h3>
    test1
  </h3>
  <div
    className="form-group"
  >
    <label
      htmlFor="1-test1-columnType"
    >
      Field type
    </label>
    <select
      className="form-control"
      disabled="true"
      id="1-test1-columnType"
      value=""
    >
      <option
        value=""
      >
        Nothing
      </option>
      <option
        key="COLUMN_NAME"
        value="COLUMN_NAME"
      >
        Persona name
      </option>
      <option
        key="COLUMN_MBOX"
        value="COLUMN_MBOX"
      >
        Identifier mbox
      </option>
      <option
        key="COLUMN_MBOXSHA1SUM"
        value="COLUMN_MBOXSHA1SUM"
      >
        Identifier mbox sha1sum
      </option>
      <option
        key="COLUMN_OPENID"
        value="COLUMN_OPENID"
      >
        Identifier open id
      </option>
      <option
        key="COLUMN_ACCOUNT_KEY"
        value="COLUMN_ACCOUNT_KEY"
      >
        Account home page
      </option>
      <option
        key="COLUMN_ACCOUNT_VALUE"
        value="COLUMN_ACCOUNT_VALUE"
      >
        Account name
      </option>
      <option
        key="COLUMN_ATTRIBUTE_DATA"
        value="COLUMN_ATTRIBUTE_DATA"
      >
        Attribute value
      </option>
    </select>
  </div>
</div>
`;
